### PR TITLE
Batch Ingress API

### PR DESCRIPTION
**Pull request by ravipilgar**
- Implement NRT JSON Batch Request API
- Implement JSON Array Request API
-Run & Test with API for batch API
-Run & Test with API for JSON Array API

### CODE REVIEW

File: Picturesbooks.csv [hunk 1]
```diff
@@ -0,0 +1,11 @@
+id,cat,name,price,inStock,author,series_t,sequence_i,genre_s
```
Comment (karthik261099) [inline Picturesbooks.csv:1]: What file is this?
This was added in previous PR too

File: Picturesbooks.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+[
```
Comment (karthik261099) [inline Picturesbooks.json:1]: What file is this?
This was added in previous PR too

File: build.gradle [hunk 3]
```diff
@@ -142,7 +142,7 @@ liquibase {
             driver "com.mysql.cj.jdbc.Driver"
             url "jdbc:mysql://localhost:3306/basicSample"
             username "root"
-            password "achhadata"
+            password "Ravikiran@123"
```
Comment (karthik261099) [inline build.gradle:145]: I think this is not needed since we are not using the database. But if an error shows up when the password is not correct, please do let us know. We'll remove the unwanted code.

File: src/main/java/com/solr/clientwrapper/infrastructure/entity/DocumentIndex.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@SolrDocument(collection = "user")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DocumentIndex {
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/infrastructure/entity/DocumentIndex.java:21]: Why is this "user" collection hardcoded...

File: src/main/java/com/solr/clientwrapper/infrastructure/entity/Employee.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.apache.solr.client.solrj.beans.Field;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.solr.core.mapping.SolrDocument;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@SolrDocument(collection = "employee")
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/infrastructure/entity/Employee.java:13]: Why is this "employee" collection hardcoded...

File: src/main/java/com/solr/clientwrapper/infrastructure/repository/EmployeeRepository.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+
+import com.solr.clientwrapper.infrastructure.entity.Employee;
+import org.springframework.data.solr.repository.Query;
+import org.springframework.data.solr.repository.SolrCrudRepository;
+
+import java.util.List;
+
+
+public interface EmployeeRepository extends SolrCrudRepository<Employee, Integer> {
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/infrastructure/repository/EmployeeRepository.java:11]: Why is this "employee" collection hardcoded...

File: src/main/java/com/solr/clientwrapper/rest/DataIngectionResource.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+	@PostMapping("/batcharray/{collectionName}")
+	@Operation(summary = "/add-document", security = @SecurityRequirement(name = "bearerAuth"))
+	public ResponseEntity<String> batcharray(@PathVariable String collectionName, @RequestBody String data) {
+
+		log.debug("json array ingection ");
+		String objectdata = dataIngectionService.parseSolrSchemaArray(collectionName, data);
+		log.debug("controller :-" + objectdata.length());
+
+		if (!objectdata.isEmpty()) {
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/rest/DataIngectionResource.java:37]: According to my review of "dataIngectionService.parseSolrSchemaArray" it always returns a toString form of JSONArray so the will always be true and the false block is never executed

File: src/main/java/com/solr/clientwrapper/rest/EmployeeController.java [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import org.springframework.web.multipart.MultipartFile;
+
+@RestController
+public class EmployeeController {
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/rest/EmployeeController.java:34]: This class was removed. Why did you add this again

File: src/main/java/com/solr/clientwrapper/rest/SolrDocParseResource.java [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+
+
+
+
+@RestController
+@RequestMapping("/file")
+public class SolrDocParseResource {
```
Comment (karthik261099) [inline src/main/java/com/solr/clientwrapper/rest/SolrDocParseResource.java:28]: This is Sprint 5's work right? Which is done by me in Sprint 6?

File: Picturesbooks.json (hunk 2 above)
Comment (ravipilgar) [reply Picturesbooks.json:1]: that file also not require.

File: build.gradle (hunk 3 above)
Comment (ravipilgar) [reply build.gradle:145]: ok

File: src/main/java/com/solr/clientwrapper/infrastructure/entity/Employee.java (hunk 5 above)
Comment (ravipilgar) [reply src/main/java/com/solr/clientwrapper/infrastructure/entity/Employee.java:13]: that class created by you

File: src/main/java/com/solr/clientwrapper/infrastructure/repository/EmployeeRepository.java (hunk 6 above)
Comment (ravipilgar) [reply src/main/java/com/solr/clientwrapper/infrastructure/repository/EmployeeRepository.java:11]: employee related class created you only

File: src/main/java/com/solr/clientwrapper/rest/EmployeeController.java (hunk 8 above)
Comment (ravipilgar) [reply src/main/java/com/solr/clientwrapper/rest/EmployeeController.java:34]: i was merge that project with my pervious repository

File: src/main/java/com/solr/clientwrapper/rest/SolrDocParseResource.java (hunk 9 above)
Comment (ravipilgar) [reply src/main/java/com/solr/clientwrapper/rest/SolrDocParseResource.java:28]: so should i delete this?

File: Picturesbooks.json (hunk 2 above)
Comment (karthik261099) [reply Picturesbooks.json:1]: Remove the file and push the changes to your branch

File: src/main/java/com/solr/clientwrapper/infrastructure/entity/Employee.java (hunk 5 above)
Comment (karthik261099) [reply src/main/java/com/solr/clientwrapper/infrastructure/entity/Employee.java:13]: I think you have created your branch from the older main branch. Please pull the new main branch and merge the main to your branch. (Do not merge your branch to main branch)

File: src/main/java/com/solr/clientwrapper/infrastructure/repository/EmployeeRepository.java (hunk 6 above)
Comment (karthik261099) [reply src/main/java/com/solr/clientwrapper/infrastructure/repository/EmployeeRepository.java:11]: I think you have created your branch from the older main branch. Please pull the new main branch and merge the main to your branch. (Do not merge your branch to main branch)

File: src/main/java/com/solr/clientwrapper/rest/SolrDocParseResource.java (hunk 9 above)
Comment (ravipilgar) [reply src/main/java/com/solr/clientwrapper/rest/SolrDocParseResource.java:28]: No Document Parse by multipart file is done By Omkar

Comment (karthik261099) [reply src/main/java/com/solr/clientwrapper/rest/SolrDocParseResource.java]: Most probably.
If this is not used anywhere else

File: src/main/java/com/solr/clientwrapper/infrastructure/entity/DocumentIndex.java (hunk 4 above)
Comment (ravipilgar) [reply src/main/java/com/solr/clientwrapper/infrastructure/entity/DocumentIndex.java]: that was old file .

File: Picturesbooks.csv (hunk 1 above)
Comment (ravipilgar) [reply Picturesbooks.csv]: I have done all changes. and push to my branch

File: src/main/java/com/solr/clientwrapper/infrastructure/entity/DocumentIndex.java (hunk 4 above)
Comment (ravipilgar) [reply src/main/java/com/solr/clientwrapper/infrastructure/entity/DocumentIndex.java]: i have changes as you commented. and push on same branch

File: src/main/java/com/solr/clientwrapper/infrastructure/entity/Employee.java (hunk 5 above)
Comment (ravipilgar) [reply src/main/java/com/solr/clientwrapper/infrastructure/entity/Employee.java]: actually i have created branch new main branch. but i thought that all code will be save on my branch so i have add all old file for reference.

File: src/main/java/com/solr/clientwrapper/domain/port/api/DataIngectionServicePort.java [hunk 10]
```diff
@@ -0,0 +1,6 @@
+package com.solr.clientwrapper.domain.port.api;
+
+public interface DataIngectionServicePort {
```
Comment (skdhitman) [inline src/main/java/com/solr/clientwrapper/domain/port/api/DataIngectionServicePort.java:3]: Small typo: it's 'j' in Injection. Would be better if we change it.
I have also renamed it for the part I have integrated your code in my branch.
